admin/module/catalog-admin-module-sources/src/main/java/org/codice/ddf/catalog/admin/plugin/AdminSourcePollerServiceBean.java

```diff
@@ -0,0 +1,280 @@
+        // Get list of metatypes
+        List<Map<String, Object>> metatypes = helper.getMetatypes();
+
+        // Loop through each metatype and find its configurations
+        for (Map metatype : metatypes) {
+            try {
+                Configuration[] configs = helper.getConfigurations(metatype);
+
+                ArrayList<Map<String, Object>> configurations = new ArrayList<>();
+                if (configs != null) {
+                    for (Configuration config : configs) {
+                        Map<String, Object> source = new HashMap<>();
+
+                        boolean disabled = config.getPid().contains(DISABLED);
+                        source.put(MAP_ENTRY_ID, config.getPid());
+                        source.put(MAP_ENTRY_ENABLED, !disabled);
+                        source.put(MAP_ENTRY_FPID, config.getFactoryPid());
+
+                        if (!disabled) {
+                            source.put(MAP_ENTRY_NAME, helper.getName(config));
+                            source.put(MAP_ENTRY_BUNDLE_NAME, helper.getBundleName(config));
+                            source.put(MAP_ENTRY_BUNDLE_LOCATION, config.getBundleLocation());
+                            source.put(MAP_ENTRY_BUNDLE, helper.getBundleId(config));
+                        } else {
+                            source.put(MAP_ENTRY_NAME, config.getPid());
+                        }
+
+                        Dictionary<String, Object> properties = config.getProperties();
+                        Map<String, Object> plist = new HashMap<>();
+                        for (String key : Collections.list(properties.keys())) {
+                            plist.put(key, properties.get(key));
+                        }
+                        source.put(MAP_ENTRY_PROPERTIES, plist);
+
+                        configurations.add(source);
+                    }
+                    metatype.put(MAP_ENTRY_CONFIGURATIONS, configurations);
+                }
+            } catch (Exception e) {
+                LOGGER.warn("Error getting source info: {}", e.getMessage());
+            }
+        }
+
+        Collections.sort(metatypes, new Comparator<Map<String, Object>>() {
+            @Override
+            public int compare(Map<String, Object> o1, Map<String, Object> o2) {
+                return ((String) o1.get("id")).compareToIgnoreCase((String) o2.get("id"));
+            }
+        });
+        return metatypes;
+    }
+
+    protected OsgiHelper getHelper() {
+        return new OsgiHelper();
+    }
+
+    public class OsgiHelper {
+        protected ConfigurationAdmin configurationAdmin;
+
+        protected BundleContext getBundleContext() {
+            Bundle bundle = FrameworkUtil.getBundle(AdminSourcePollerServiceBean.class);
+            if (bundle != null) {
+                return bundle.getBundleContext();
+            }
+            return null;
+        }
+
+        protected List<ServiceReference<? extends Source>> getServiceReferences()
```

Comment:
You should be able to hide more OSGi details to make testing and mocking easier.  Instead of returning service references, you could have a method that returns sources that are configured sources that match a given service.pid.